lib/LWP/UserAgent.pm

```diff
@@ -1039,6 +1043,10 @@ sub mirror
             rename( $tmpfile, $file )
                 or die "Cannot rename '$tmpfile' to '$file': $!\n";
 
+            my $mode = 0666 &~ (umask() || 0);
```

Comment:
I'm undecided if 0666 is good when umask is not supported (== returns undef), or if we should use some other perms (0644?) or skip chmod altogether when that's the case.

Reply:
Any thoughts on this?

Reply:
I wonder if skipping `chmod` would be the least confusing thing to do when `umask` is not supported.

Reply:
For better or for worse, that'll leave us with whatever was set for the tmpfile. Do we have anyone listening here who would have access to a system that doesn't support `umask`, and could confirm what do file permissions for tempfiles created the way we do here look like?

One thing we could do is to copy over the mode bits of the previous file if it existed, and not set ones to something we decide every time. That won't solve the problem discussed here, but would make sense to me. But then again, should we do this even if the previous file wasn't owned by us?

Reply:
I think for existing files, #396 is the correct solution.  Do note that [File::Temp has a `PERMS` flag](https://metacpan.org/pod/File::Temp) that seems to handle at least a few different operating systems.

Reply:
Oh, I had already forgotten filing #396, thanks for the reminder! I'll skip preserving existing perms for the purposes of this PR so we don't balloon it any more. 53f5c63 skips the `chmod` if `umask` is not supported.

Reply:
0666 & umask seems like it gives the user the most control.  Should probably be documented that it does that though.

Reply:
I don't think that is correct use of umask. For example with a standard 0002 one, that gives `--------w-`, whereas `0666 &~ umask` as in this PR currently gives the expected `-rw-rw-r--`.

Reply:
Yes, what you have.  I meant "0666 with umask" and was not careful in my wording.

Reply:
I wouldn't bother documenting it though; that's the way commands that write files are expected to work. If they do something different, then _that_ I think should be documented.